File: Stripe/STPPaymentMethodSEPADebitParams.m
//
//  STPPaymentMethodSEPADebitParams.m
//  StripeiOS
//
//  Created by Cameron Sabol on 10/7/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

#import "STPPaymentMethodSEPADebitParams.h"

NS_ASSUME_NONNULL_BEGIN

@implementation STPPaymentMethodSEPADebitParams

@synthesize additionalAPIParameters = _additionalAPIParameters;

#pragma mark - STPFormEncodable

+ (nullable NSString *)rootObjectName {
    return @"sepa_debit";
}

+ (NSDictionary *)propertyNamesToFormFieldNamesMapping {
    return @{
             NSStringFromSelector(@selector(iban)): @"iban",
             };
}
@end

NS_ASSUME_NONNULL_END

File: Stripe/STPPaymentMethodSEPADebitParams.swift
//
//  STPPaymentMethodSEPADebitParams.swift
//  StripeiOS
//
//  Created by Cameron Sabol on 10/7/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

import Foundation

/// An object representing parameters used to create a SEPA Debit Payment Method
public class STPPaymentMethodSEPADebitParams: NSObject, STPFormEncodable {
  @objc public var additionalAPIParameters: [AnyHashable: Any] = [:]

  /// The IBAN number for the bank account you wish to debit. Required.
  @objc public var iban: String?

  // MARK: - STPFormEncodable
  @objc
  public class func rootObjectName() -> String? {
    return "sepa_debit"
  }

  @objc
  public class func propertyNamesToFormFieldNamesMapping() -> [String: String] {
    return [
      NSStringFromSelector(#selector(getter:iban)): "iban"
    ]
  }
}
